test/temporal_terms.jl

```diff
@@ -0,0 +1,105 @@
+using StatsModels
+using StatsBase
+using StatsModels: width
+using DataStructures
+
+@testset "Temporal Terms" begin
+    @testset "Lag" begin
+        @testset "Basic use" begin
+            df = (y=1:10, x = 1:10)
+            f = @formula(y ~ lag(x, 0) + lag(x, 1) + lag(x, 3) + lag(x, 11))
```

Comment:
Should negative values work? Better test that too.

Reply:
That is tested below